src/main/java/uk/ac/cam/cl/dtg/segue/api/UsersFacade.java

```diff
@@ -614,16 +614,15 @@ public Response getAllSchoolOtherResponses(@Context final Request request) {
      *            - the new user object from the clients perspective.
      * @param passwordCurrent
      * 			  - the current password, used if the password has changed
-     * @param emailPreferences
-     * 			  - the email preferences for this user
-     * @param subjectInterests - the subjects interests of the user, which should be removed from this method!
+     * @param userPreferences
+     * 			  - the preferences for this user
      * @return the updated user object.
      * @throws NoCredentialsAvailableException
      * @throws IncorrectCredentialsProvidedException
      */
     private Response updateUserObject(final HttpServletRequest request, final HttpServletResponse response,
                                       final RegisteredUser userObjectFromClient, final String passwordCurrent,
-                                      final List<IEmailPreference> emailPreferences, final Map<String, Boolean> subjectInterests)
+                                      final Map<String, Map<String, Boolean>> userPreferences)
```

Comment:
This looks good for now, but I would like to see the updateUserObject method simplified in the future. As it isn't just the user object now. Perhaps that will be my next mission..